polynote-frontend/polynote/ui/component/notebook/notebookconfig.ts

```diff
@@ -401,15 +401,33 @@ class SparkConf {
 
 }
 
-class EnvConf {
+class KernelConf {
     readonly el: TagElement<"div">;
     private container: TagElement<"div">;
+    private jvmArgsInput: TagElement<"input">;
+    private scalaVersionInput: DropdownElement;
+
+    constructor(configState: StateView<NotebookConfig>) {
+        const envHandler = configState.view("env");
+        const jvmArgsHandler = configState.view("jvmArgs");
+        const scalaVersionHandler = configState.view("scalaVersion");
+
+        // TODO: this could come from the server
+        const availableScalaVersions = {
+            "2.11": "2.11",
+            "2.12": "2.12"
+        }
 
-    constructor(envHandler: StateView<Record<string, string> | undefined>) {
         this.el = div(['notebook-env', 'notebook-config-section'], [
-            h3([], ['Environment Variables']),
-            para([], ['Set environment variables here. Please note this is only supported when kernels are launched as a subprocess (default).']),
-            this.container = div(['env-list'], [])
+            h3([], ['Kernel configuration']),
+            para([], ['Please note this is only supported when kernels are launched as a subprocess (default).']),
+            h4([], 'Scala version:'),
+            para([], `If using Spark, the Scala version must match that of your Spark installation. "Default" will use Polynote's configured Scala version, or auto-detect the appropriate version.`),
```

Comment:
* If an explicit (not "Default") Scala version is configured in the notebook, it will use that.
* Otherwise, if polynote is configured with `kernel.scala_version` then it will use that.
* Otherwise, if you're launching a Spark kernel, it will try to detect which Scala version your Spark installation wants.
* Otherwise, it will use the same Scala version that the running server was built with.

Reply:
In that case, maybe this would be more in line with the order of operations you're describing?
```suggestion
            para([], `If using Spark, the Scala version must match that of your Spark installation. "Default" will auto-detect the appropriate version if possible, otherwise falling back to using Polynote's configured Scala version`),
```